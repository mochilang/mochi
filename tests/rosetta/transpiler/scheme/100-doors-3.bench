{
  "duration_us": 142,
  "memory_bytes": 12697600,
  "name": "main"
}