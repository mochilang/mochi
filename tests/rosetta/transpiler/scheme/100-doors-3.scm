;; Generated on 2025-07-25 08:58 +0700
(import (only (scheme base) call/cc when list-ref list-set! list))
(import (scheme time))
(import (chibi string))
(import (only (scheme char) string-upcase string-downcase))
(import (chibi time) (srfi 98))
(define _now_seeded #f)
(define _now_seed 0)
(define (now)
  (when (not _now_seeded)
    (let ((s (get-environment-variable "MOCHI_NOW_SEED")))
      (when (and s (string->number s))
        (set! _now_seed (string->number s))
        (set! _now_seeded #t))))
  (if _now_seeded
      (begin
        (set! _now_seed (modulo (+ (* _now_seed 1664525) 1013904223) 2147483647))
        _now_seed)
      (* (current-seconds) 1000000000)))(import (chibi time))
(define (_mem) (* 1024 (resource-usage-max-rss (get-resource-usage resource-usage/self))))
(import (chibi json))
(define (to-str x)
  (cond ((pair? x)
         (string-append "[" (string-join (map to-str x) ", ") "]"))
        ((string? x) x)
        ((boolean? x) (if x "1" "0"))
        (else (number->string x))))
(define (upper s) (string-upcase s))
(define (lower s) (string-downcase s))
(define (fmod a b) (- a (* (floor (/ a b)) b)))
(let ((start5 (now)
)
)
 (begin (define result "")
 (call/cc (lambda (break2)
 (letrec ((loop1 (lambda (i)
 (if (< i 101)
 (begin (begin (let ((j 1)
)
 (begin (call/cc (lambda (break4)
 (letrec ((loop3 (lambda ()
 (if (< (* j j)
 i)
 (begin (set! j (+ j 1)
)
 (loop3)
)
 (quote nil)
)
)
)
)
 (loop3)
)
)
)
 (if (equal? (* j j)
 i)
 (begin (set! result (string-append result "O")
)
)
 (begin (set! result (string-append result "-")
)
)
)
)
)
)
 (loop1 (+ i 1)
)
)
 (quote nil)
)
)
)
)
 (loop1 1)
)
)
)
 (display (to-str result)
)
 (newline)
 (let ((end6 (now)
)
)
 (let ((dur7 (quotient (- end6 start5)
 1000)
)
)
 (begin (display (string-append "{\n  \"duration_us\": " (number->string dur7)
 ",\n  \"memory_bytes\": " (number->string (_mem)
)
 ",\n  \"name\": \"main\"\n}")
)
 (newline)
)
)
)
)
)
