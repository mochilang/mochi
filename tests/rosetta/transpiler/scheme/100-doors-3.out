O--O----O------O--------O----------O------------O--------------O----------------O------------------O
{
  "duration_us": 571223,
  "memory_bytes": 13033472,
  "name": "main"
}